src/test/debugger/android/androidPlatform.test.ts

```diff
@@ -0,0 +1,280 @@
+                        // It should be running in exactly one of these two devices
+                        isRunningOnNexus5.should.not.eql(isRunningOnNexus6);
+                        const emulatorWithAppRunningId = isRunningOnNexus5 ? "Nexus_5" : "Nexus_6";
+                        shouldHaveReceivedSingleLogCatMessage(emulatorWithAppRunningId);
+                    });
+            });
+
+        testWithRecordings("runApp launches the app when three emulators are connected",
+            ["react-native/run-android/win10-rn0.21.0/succeedsWithThreeVSEmulators"], () => {
+                return Q({})
+                    .then(() => {
+                        return simulatedAVDManager.createAndLaunchAll("Nexus_5", "Nexus_6", "Other_Nexus_6");
+                    }).then(() => {
+                        return androidPlatform.runApp();
+                    }).then(() => {
+                        return Q.all([
+                            deviceHelper.isAppRunning(androidPackageName, "Nexus_5"),
+                            deviceHelper.isAppRunning(androidPackageName, "Nexus_6"),
+                            deviceHelper.isAppRunning(androidPackageName, "Other_Nexus_6"),
+                        ]);
+                    }).then(isRunningList => {
+                        // It should be running in exactly one of these two devices
+                        isRunningList.filter(v => v).should.eql([true]);
+
+                        // Get index of running emulator
+                        const index = isRunningList.indexOf(true);
+                        const emulatorWithAppRunningId = ["Nexus_5", "Nexus_6", "Other_Nexus_6"][index];
+                        shouldHaveReceivedSingleLogCatMessage(emulatorWithAppRunningId);
+                    });
+            });
+
+        testWithRecordings("runApp fails if no devices are connected",
+            ["react-native/run-android/win10-rn0.21.0/failsDueToNoDevicesConnected"], () => {
+                return Q({})
+                    .then(() => {
+                        return androidPlatform.runApp();
+                    }).then(() => {
+                        should.assert(false, "runApp should've exited with an error");
+                    }, reason => {
+                        reason.message.should.eql("Unknown error");
+                        shouldHaveReceivedNoLogCatMessages();
+                    });
+            });
+
+        testWithRecordings("runApp launches the app in an online emulator only",
+            ["react-native/run-android/win10-rn0.21.0/succeedsWithFiveVSEmulators"], () => {
+                return Q({})
+                    .then(() => {
+                        return simulatedAVDManager.createAndLaunchAll("Nexus_5", "Nexus_6", "Nexus_10", "Nexus_11", "Nexus_12");
+                    }).then(() => {
+                        return deviceHelper.notifyDevicesAreOffline("Nexus_5", "Nexus_6", "Nexus_10", "Nexus_12");
```

Comment:
Same comment about making this method take an array instead. Then the tests can use something like
```
let offlineDevices = ["Nexus_5", "Nexus_6"];
let onlineDevices = ["Nexus_10", "Nexus_11"];

...simulatedAVDManager.createAndLaunchAll(offlineDevices.concat(onlineDevices))
...simulatedAVDManager.notifyDevicesAreOffline(offlineDevices)
```